datasets.py

```diff
@@ -117,6 +122,89 @@ def read_xml_content(xml_file):
     11: 'indoor'
 }
 
+def group_mapping_creator(labels_to_names, supercategories_to_names, 
+                          override_map = None):
+    '''
+    inputs:
+    labels_to_names: dict mapping "label" to "human readable string"
+    supercategories_to_names: dict mapping "supercat" to "human readable string"
+    override_map: dict mapping: "human-readable label" to "human-readable supercat"
+                        for manual overriding of mapping of certain labels
+           
+    output:
+    prints out human-readable label to human readable supercat mapping
+    
+    returns:
+    function that takes an input label, and returns a supercat
+    '''
+    ######################################################################  
+    assert (labels_to_names is not None and supercategories_to_names is not None)
+    # function that calculates the "closest" supercategory to input word
+    # and returns said supercategory and the associated softmax confidence score
+    def dist_calculator(word):
+        assert(word is not None)
+        # create list of supercats for easy indexing & ordering
+        supercat_list = list(supercategories_to_names.keys())
+        # score array for each supercat, for softmax calculation
+        score_arr = []
+
+        for supercat in (supercat_list):
+            word1 = nlp(supercategories_to_names[supercat])
+            word2 = nlp(word)
+            cur_score = word1.similarity(word2)
+            if cur_score is None:
```

Comment:
when does this happen?

Reply:
Thanks -- I think you're referring to the "cur_score is None" check. I took a look at the documentation and the similarity() function always returns a float, so I removed the None check. (Let me know if this is what you were referring to?)


Also, unrelated I think but in line 152, I'm needlessly computing the nlp tokens for the supercats each time the dist_calculator() function is being run, so I precomputed the tokens outside of the function.